### ISSUE TITLE
On ppc64, malloc gives misaligned buffers for glibc <= 2.23.

### ISSUE THREAD

**Issue by ahaldane** [numpy/numpy#10491]
On ppc64 and ppc64le systems, glibc 2.23-or-less has a bug in malloc, that it only guarantees alignment to 8 bytes, even though it should be 16 bytes.

https://sourceware.org/bugzilla/show_bug.cgi?id=6527

This is a problem for the long-double (float128) and sometimes complex128 type which should be aligned to 16 bytes. This means that when using long-double arrays, many internal numpy computations will randomly fail on ppc64, depending on what alignment malloc returned. This causes sporadic numpy unit test failures and sometimes outright crashes. The test `numpy.core.tests.test_numeric TestRequire.test_require_each` commonly fails, though not consistently.

Users on ppc64 should upgrade to glibc 2.24 or later.

(Discovered in #10443)

**Comment by jjhelmus** [comment]
Does adding `-malign-natural` to the compiler flags have any effect on the alignment?

**Comment by ahaldane** [comment]
Nope, I still get the error on glibc 2.23, `C compiler: powerpc-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -malign-natural -fPIC` (gcc 5.4.0).

Here is a test script I am running:
```python
import sys
c = []
for i in range(1000):
    c.append(np.zeros(np.random.randint(10), dtype='f16'))
    if (c[-1].ctypes.data % 16) != 0:
        print("error:", c[-1].flags['A'], c[-1].ctypes.data)
        break
```
which shows I get 8-byte-only alignment sometimes. I also confirm `test_require_each` in the test suite still fails sometimes.

**Comment by mattip** [comment]
We now have CI for ppc64. Closing. Please open a new issue as needed.
